pkg/expr/nodes.go

```diff
@@ -247,6 +248,7 @@ func (dn *DSNode) Execute(ctx context.Context, now time.Time, _ mathexp.Vars, s
 
 	var result mathexp.Results
 	responseType, result, err = queryDataResponseToResults(ctx, resp, dn.refID, dn.datasource.Type, s)
+	makeResultsIdentifiableByLabels(result)
```

Comment:
I think this should be called within `queryDataResponseToResults` (towards the end). While we are looping over the frames:

```
for _, frame := range response.Frames {
		// Check for TimeSeriesTypeNot in InfluxDB queries. A data frame of this type will cause
		// the WideToMany() function to error out, which results in unhealthy alerts.
		// This check should be removed once inconsistencies in data source responses are solved.
		if frame.TimeSeriesSchema().Type == data.TimeSeriesTypeNot && datasourceType == datasources.DS_INFLUXDB {
			logger.Warn("Ignoring InfluxDB data frame due to missing numeric fields")
			continue
		}
		var series []mathexp.Series
		series, err := WideToMany(frame)
		if err != nil {
			return "", mathexp.Results{}, err
		}
		for _, ser := range series {
			vals = append(vals, ser)
		}
	}
```

we can have a bool to track if there are no labels for all series, and if that is true (and more than 1 series or whatever) then we call the function to create `__name__`. 

So the validation at the top of makeResultsIdentifableByLabels will move outside of the function.